sample-data/fake.js

```diff
@@ -2,119 +2,148 @@ let faker = require('faker');
-        house.houseType = houseTypes[Math.floor(Math.random() * houseTypes.length)];
-        house.yearBuilt = Math.floor(Math.random() * 88 + 1930);
-        house.heating = heatingTypes[Math.floor(Math.random() * heatingTypes.length)];
-        house.cooling = coolingTypes[Math.floor(Math.random() * coolingTypes.length)];
-        house.parking = Math.floor(Math.random() * 6 + 1);
-        house.lotSize = Math.floor(Math.random() * (10000 - house.floorSize) + house.floorSize);
-        house.daysListed = Math.floor(Math.random() * 364 + 1);
-        house.saves = Math.floor(Math.random() * 150);
-        
-        sampleData.push(house);
-        
-        ok = writeStream.write(csvStringifier.stringifyRecords(sampleData), 'utf8');
-        
-        sampleData = [];
-        
-        i--;
-        
-        if(i % 1e4 === 0){
-            console.log(i);
-        } else if (i % 1e5 === 0) {
-            console.clear();
+        const createCsvStringifier = require('csv-writer').createObjectCsvStringifier;
+        const csvStringifier = createCsvStringifier({
+            path: path.join(__dirname, 'data.csv'),
+            header: [
+                {id: 'appliances', title: 'APPLIANCES'},
+                {id: 'interiorFeatures', title: 'INTERIORFEATURES'},
+                {id: 'construction', title: 'CONSTRUCTION'},
+                {id: 'roof', title: 'ROOF'},
+                {id: 'exterior', title: 'EXTERIOR'},
+                {id: 'flooring', title: 'FLOORING'},
+                {id: 'homeId', title: 'ID'},
+                {id: 'homeAddress', title: 'ADDRESS'},
+                {id: 'price', title: 'PRICE'},
+                {id: 'beds', title: 'BEDS'},
+                {id: 'baths', title: 'BATHS'},
+                {id: 'rooms', title: 'ROOMS'},
+                {id: 'stories', title: 'STORIES'},
+                {id: 'floorSize', title: 'FLOORSIZE'},
+                {id: 'spaces', title: 'SPACES'},
+                {id: 'houseDescription', title: 'DESCRIPTION'},
+                {id: 'houseType', title: 'TYPE'},
+                {id: 'yearBuilt', title: 'YEAR'},
+                {id: 'heating', title: 'HEATING'},
+                {id: 'cooling', title: 'COOLING'},
+                {id: 'parking', title: 'PARKING'},
+                {id: 'lotSize', title: 'LOTSIZE'},
+                {id: 'daysListed', title: 'DAYSLISTED'},
+                {id: 'saves', title: 'SAVES'}
+            ],
+            fieldDelimiter: ';'
+        });
+
+        let i = recordQuantity;
+
+        write();
+
+        function write() {
+            i--;
+            ok = true;
+            do {
+                let streetAddress = faker.address.streetAddress();
+                let streetName = faker.address.streetName();
+                let city = faker.address.city();
+                let state = faker.address.state();
+                let zip = faker.address.zipCode();
```

Comment:
Most of these can be initialized with `const` since they are not reassigned before the end of the `do { }` block, and will go out of scope.